scri/asymptotic_bondi_data/transformations.py

```diff
@@ -40,7 +40,7 @@ def _process_transformation_kwargs(input_ell_max, **kwargs):
                 i_neg = sf.LM_index(ell, -m, 0)
                 a = supertranslation[i_pos]
                 b = supertranslation[i_neg]
-                if abs(a - (-1.0) ** m * b.conjugate()) > 3e-16 + 1e-15 * abs(b):
+                if abs(a - (-1.0) ** m * b.conjugate()) > 1e-12 + 1e-12 * abs(b):
```

Comment:
IIRC, I chose these tolerances based on my experience, where I would explicitly impose symmetries.  E.g., I might just define the non-negative modes, and then fill in negative modes based on the reality requirement.

I worry about unexpected wierdness if the supertranslation is partially imaginary, even at the 1e-12 level.  I imagine somebody trying to diagnose a problem somewhere down the line, overlooking the cause because they quite reasonably assume supertranslations are always real.

Would it be better to arbitrarily just choose a way to impose reality?  Maybe we could take the appropriately signed "average" of the two modes and just reset each value?  If 1e-12 errors are not significant, what problems could it cause to impose reality like this?

Reply:
Yeah I think like the idea of a signed average of the two modes. I can try that and see if the tolerances are obeyed.